src/components/App/ExpressRecorder.tsx

```diff
@@ -667,10 +705,12 @@ export class ExpressRecorder extends Component<ExpressRecorderProps, State> {
                         onUploadCancelled={this.handleUploadCancelled}
                         onUploadProgress={this.handleUploadProgress}
                         mediaType={
-                            constraints.video ? KalturaMediaType.video : KalturaMediaType.audio
+                            constraints.video || shareScreenOn
+                                ? KalturaMediaType.video
+                                : KalturaMediaType.audio
                         }
-                        recordedBlobs={recordedBlobs}
-                        childRecordedBlobs={screenRecordedBlobs}
+                        recordedBlobs={recordedBlobs.length ? recordedBlobs : screenRecordedBlobs}
+                        childRecordedBlobs={recordedBlobs.length ? screenRecordedBlobs : undefined}
```

Comment:
I suggest creating some variable before the component returns an JSX component.
please add on line 689 the following:
```
const uploadManagerProps: {
  mediaType: constraints.video || shareScreenOn ? KalturaMediaType.video : KalturaMediaType.audio,
  recordedBlobs: recordedBlobs.length ? recordedBlobs : screenRecordedBlobs,
  ...
}
```
and after you may just send props with spread operator:
```
<UploadManager 
  ...existing props
  {...uploadManagerProps}
/>
```

Reply:
why? it's moving a mess from 1 place to another as is..

Reply:
it's not just for sending props to another component.
it's a way to code as narrow and readable code as possible.

when you define your props from outside the component call:
1. you can aggregate your variables to a single source-of-truth object.
2. IMO, the spread operator may be used to spreading the relevant HTML attributes and component props with a single and short line of code - which is great.

Reply:
ok, done